GPU/GPUCommon.cpp

```diff
@@ -1677,15 +1722,22 @@ void GPUCommon::Execute_TgenMtxNum(u32 op, u32 diff) {
 	const int end = 12 - (op & 0xF);
 	int i = 0;
 
-	while ((src[i] >> 24) == GE_CMD_TGENMATRIXDATA) {
-		const u32 newVal = src[i] << 8;
-		if (dst[i] != newVal) {
-			Flush();
-			dst[i] = newVal;
-			gstate_c.Dirty(DIRTY_TEXMATRIX);
-		}
-		if (++i >= end) {
-			break;
+	bool skipData = debugRecording_;
```

Comment:
Worth noting: this will skip certain optimizations, like CALL.  It might be possible to preserve those in some way, but right now it's just executing all commands the GE runs - which keeps it simple.

It becomes tricky to manage when games are updating stalls, etc. and recording changes correctly - you get to a point where you're basically trying to recreate save states.

-[Unknown]